File: src/main/java/Funding/Startreum/domain/comment/exception/CommentExceptionHandler.java
package Funding.Startreum.domain.comment.exception;

import Funding.Startreum.common.util.ApiResponse;
import Funding.Startreum.domain.comment.controller.CommentRestController;
import jakarta.persistence.EntityNotFoundException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice(assignableTypes = CommentRestController.class)
public class CommentExceptionHandler {

    @ExceptionHandler(EntityNotFoundException.class)
    public ResponseEntity<ApiResponse<Void>> handleEntityNotFoundException(EntityNotFoundException e) {
        return ResponseEntity
                .status(HttpStatus.NOT_FOUND)
                .body(ApiResponse.error(e.getMessage()));
    }

}

File: src/main/java/Funding/Startreum/domain/comment/exception/CommentExceptionHandler.kt
package funding.startreum.domain.comment.exception

import funding.startreum.common.util.ApiResponse
import funding.startreum.domain.comment.controller.CommentRestController
import jakarta.persistence.EntityNotFoundException
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice(assignableTypes = [CommentRestController::class])
class CommentExceptionHandler {
    @ExceptionHandler(EntityNotFoundException::class)
    fun handleEntityNotFoundException(e: EntityNotFoundException): ResponseEntity<ApiResponse<Void>> {
        return ResponseEntity
            .status(HttpStatus.NOT_FOUND)
            .body(ApiResponse.error(e.message))
    }
}
